analyzers/src/SonarAnalyzer.CSharp/Rules/DatabasePasswordsShouldBeSecure.cs

```diff
@@ -103,13 +104,13 @@ private void CheckAppSettingJson(CompilationAnalysisContext c, string fullPath)
             var appSettings = File.ReadAllText(fullPath);
             if (appSettings.Contains("\"ConnectionStrings\""))
             {
-                using var jsonReader = new JsonTextReader(new StringReader(appSettings));
                 try
                 {
-                    var json = JObject.Load(jsonReader, new JsonLoadSettings {LineInfoHandling = LineInfoHandling.Load, CommentHandling = CommentHandling.Ignore});
+                    var analyzer = new SyntaxAnalyzer(appSettings);
+                    var json = analyzer.Parse();
```

Comment:
We can remove these two lines if we embedthe `JsonNode.FromString(appSettings)` in the `ReportEmptyPassword`.